cli/src/firebase_management_rest_service.py

```diff
@@ -0,0 +1,286 @@
+                         "{database_id}")
+
+# There is a databaseId parameter that will be added, but is not included here
+RTDB_INSTANCE_CREATE_URL = ("https://firebasedatabase.googleapis.com/v1beta/"
+                            "projects/{project_id}/locations/{location}/"
+                            "instances")
+
+PROJECTS_GET_STATE_ERROR_MSG = """
+  ERROR. Expected to find the 'state' field populated in the projects get
+  response with a value of 'ACTIVE'. However the response was the following:
+
+  {response}
+"""
+
+RTDB_INSTANCE_GET_STATE_ERROR_MSG = """
+  ERROR. Expected to find the 'state' field populated in the database instance
+  response with a value of 'ACTIVE'. However the response was the following:
+
+  {response}
+"""
+
+
+class FirebaseManagementRestService:
+  """Implements a service for making Firebase RTDB management REST requests.
+
+  This service is for making management/admin related Firebase RTDB
+  requests/queries using the REST interface, which is documented at
+  https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects/.
+  """
+
+  def __init__(self, http_service, project_id, user_output):
+    self.http_service = http_service
+    self.project_id = project_id
+    self.user_output = user_output
+
+  def project_get(self):
+    """Checks if the project is enabled for firebase.
+
+      Returns:
+        A value of type FirebaseProjectStatus representing the project's status.
+    """
+
+    # On success, a response like the following is expected:
+    # {
+    #   "projectId": "some-project",
+    #   "projectNumber": "<numeric id>",
+    #   "displayName": "jcborg-work3-test1",
+    #   "name": "projects/some-project,
+    #   "resources": {
+    #     "hostingSite": "some-project"
+    #   },
+    #   "state": "ACTIVE"
+    # }
+    #
+    # Errors (That we handle):
+    #   404 NOT_FOUND: If the project has not been enabled for Firebase yet.
+
+    # Command documentation at:
+    # https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects/get
+
+    url = PROJECTS_GET_URL.format(project_id=self.project_id)
+    request = self.http_service.build_request(
+        "GET", url, include_project_header=True)
+    response = None
+
+    try:
+      response = self.http_service.send(request, handle_http_error=False)
+    except HTTPError as err:
+      if err.code == 404:
+        self.user_output.debug("Got 404, project did not exist")
+        return FirebaseProjectStatus.NOT_ENABLED
+
+      print_http_error(self.user_output, request, err)
+      raise SilentlyExitError from err
+
+    if "state" not in response or response["state"] != "ACTIVE":
+      # The documention doesn't specify all the values for state.
+      #
+      # However for the database instance command:
+      # https://firebase.google.com/docs/reference/rest/database/database-management/rest/v1beta/projects.locations.instances#state
```

Comment:
:)  Loving the links to supporting documentation.

Reply:
Thansk, yeah, they are definitely handy to have.